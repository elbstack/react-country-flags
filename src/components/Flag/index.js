import React, { Component, createElement } from 'react';
import PropTypes from 'prop-types';
import flags from '../../flags/react';

const fetchFlag = async country => {
  if (!flags[country]) return Promise.resolve(null);
  const { default: data } = await flags[country]();
  return data;
};

export const countries = (() => Object.keys(flags))();

export default class Flags extends Component {
  state = {
    flag: null,
  };

  static defaultProps = {
    country: null,
    asSquare: false,
  };

  componentDidMount() {
    this.updateFlag();
  }

  componentDidUpdate(prevProps) {
    const { country } = this.props;

    if (country !== prevProps.country) {
      this.updateFlag();
    }
  }

  async updateFlag() {
    const { country } = this.props;
    const flag = await fetchFlag(country);
    this.setState({ flag });
  }

  render() {
    const { asSquare } = this.props;
    const { flag } = this.state;
    if (!flag) return null;
    if (!asSquare) return createElement(flag);
    return (
      <svg viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
        {createElement(flag)}
      </svg>
    );
  }
}

Flags.propTypes = {
  country: PropTypes.string,
  asSquare: PropTypes.bool,
};
