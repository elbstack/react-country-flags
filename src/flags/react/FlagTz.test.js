import React from 'react';
import renderer from 'react-test-renderer';
import Flag from './FlagTz';

it('FlagTz renders without crashing', () => {
  const component = renderer.create(<Flag />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
