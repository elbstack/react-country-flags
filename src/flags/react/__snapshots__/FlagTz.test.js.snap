// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagTz renders without crashing 1`] = `
<svg
  viewBox="0 0 72 48"
>
  <svg
    viewBox="0 0 72 48"
  >
    <path
      d="M0 48V0h72z"
      fill="#1eb53a"
    />
    <path
      d="M0 48h72V0z"
      fill="#00a3dd"
    />
    <path
      d="M0 48L72 0"
      stroke="#fcd116"
      strokeWidth={19}
    />
    <path
      d="M0 48L72 0"
      stroke="#000"
      strokeWidth={13}
    />
  </svg>
</svg>
`;
